src/App.tsx

```diff
@@ -7,33 +9,80 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  time: string;
+  timerName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    time: '',
+    timerName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
```

Comment:
You should add a listener to the componentDidMount to hide the clock on the right mouse click. Now it's not implemented, but required in the task